echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/slack/SlackService.groovy

```diff
@@ -17,27 +17,104 @@
 
 package com.netflix.spinnaker.echo.slack
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.netflix.spinnaker.echo.config.SlackConfig
+import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException
 import groovy.json.JsonBuilder
 import groovy.transform.Canonical
+import groovy.util.logging.Slf4j
+import org.springframework.http.HttpHeaders
+import org.springframework.http.RequestEntity
+import org.springframework.security.crypto.codec.Hex
 import retrofit.client.Response
 
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+import java.security.InvalidKeyException
+import java.time.Duration
+import java.time.Instant
+
 @Canonical
+@Slf4j
 class SlackService {
-
   SlackClient slackClient
-  boolean useIncomingWebHook
+  SlackConfig.SlackProperties configProperties
+
+  Response sendCompactMessage(CompactSlackMessage message, String channel, boolean asUser) {
+    slackClient.sendMessage(configProperties.token, message.buildMessage(), channel, asUser)
+  }
 
-  Response sendCompactMessage(String token, CompactSlackMessage message, String channel, boolean asUser) {
-    slackClient.sendMessage(token, message.buildMessage(), channel, asUser)
+  Response sendMessage(SlackAttachment message, String channel, boolean asUser) {
+    configProperties.useIncomingWebhook ?
+      slackClient.sendUsingIncomingWebHook(configProperties.token, new SlackRequest([message], channel)) :
+      slackClient.sendMessage(configProperties.token, toJson(message), channel, asUser)
```

Comment:
Is the slack token constant, or does it change periodically? Wondering if this token is like a github token which is generated once, or a more message-specific token

Reply:
Permanent (at least until you manually rotate it in the Slack admin UI). This was already configured for our deployment.